internal/application/command.go

```diff
@@ -34,23 +34,23 @@ type CommandProcessor struct {
 	device        models.Device
 	sourceName    string
 	correlationID string
-	body          string
+	setParamsMap  map[string]string
 	attributes    string
 	dic           *di.Container
 }
 
-func NewCommandProcessor(device models.Device, sourceName string, correlationID string, body string, attributes string, dic *di.Container) *CommandProcessor {
+func NewCommandProcessor(device models.Device, sourceName string, correlationID string, setParamsMap map[string]string, attributes string, dic *di.Container) *CommandProcessor {
```

Comment:
if setParamsMap is nil, initialize it as an empty map to prevent any nil pointer issue